test/Npgsql.Tests/AuthenticationTests.cs

```diff
@@ -281,8 +281,8 @@ public void Password_source_precedence()
             builder.Passfile = passFile;
             builder.ApplicationName = $"{nameof(Password_source_precedence)}:{Guid.NewGuid()}";
 
-            using var pool = CreateTempPool(builder.ConnectionString, out var connectionString);
-            using var connection = await OpenConnectionAsync(connectionString);
+            await using var dataSource = CreateDataSource(builder);
```

Comment:
This looks wrong, since dispose happens at the end of the local method (it was wrong before as well)